habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -216,30 +220,47 @@ def get_walk_pose(
         joint_pose, obj_transform = new_pose.joints, new_pose.root_transform
 
         # We correct the object transform
-        obj_transform.translation *= 0
         look_at_path_T = mn.Matrix4.look_at(
-            self.obj_transform.translation,
-            self.obj_transform.translation + forward_V.normalized(),
+            self.obj_transform_base.translation,
+            self.obj_transform_base.translation + forward_V.normalized(),
             mn.Vector3.y_axis(),
         )
 
         # Remove the forward component, and orient according to forward_V
         obj_transform.translation *= mn.Vector3.x_axis() + mn.Vector3.y_axis()
-        obj_transform = look_at_path_T @ obj_transform
+        obj_transform_offset = obj_transform
 
-        obj_transform.translation += (
-            forward_V * dist_diff * distance_multiplier
-        )
-        self.obj_transform = obj_transform
+        # This is the rotation and translation caused by the current motion pose
+        #  we still need to apply the base_transform to obtain the full transform
+        self.obj_transform_offset = obj_transform
+
+        # The base_transform here is independent of transforms caused by the current
+        # motion pose.
+        obj_transform_base = look_at_path_T
+        forward_V_dist = forward_V * dist_diff * distance_multiplier
+        obj_transform_base.translation += forward_V_dist
+
+        self.obj_transform_base = obj_transform_base
 
-        return joint_pose, obj_transform
+        return joint_pose, obj_transform_offset, obj_transform_base
 
     @classmethod
-    def vectorize_pose(cls, pose: List, transform: mn.Matrix4):
+    def vectorize_pose(
```

Comment:
This is a bit awkward. I wonder if having `get_xxx_pose`  should be renamed to `calculated_xxx_pose` and if instead of returning the joint_pose, obj_transform_offset, obj_transform_base it should return nothing and then have `get_vectorize_pose` (rename to `get_full_pose` ? ) and `get_pose` return the pose. It is strange to return these variables that will then be put back as inputs.